frontend/src/contexts/modalContext.tsx

```diff
@@ -0,0 +1,53 @@
+import { ModalProps } from 'components/common/GenericModal';
+import noop from 'lodash/noop';
+import * as React from 'react';
+import { useState } from 'react';
+
+export interface IModalContext {
+  modalProps?: ModalProps;
+  setModalProps: (modalProps?: ModalProps) => void;
+  setDisplayModal: (display: boolean) => void;
+}
+
+export const ModalContext = React.createContext<IModalContext>({
+  modalProps: undefined,
+  setModalProps: noop,
+  setDisplayModal: noop,
```

Comment:
If is not set and gets called, should it throw an error or log something instead?

Reply:
sure, couldn't hurt. They are always being set currently but would be nice to throw an error if someone messes this up.